### PR TITLE
Updating CreateScalar to be intrinsic for Vector64/128/256

### PR DESCRIPTION
**Pull request by tannergooding**
This ensures that vector constant nodes can be created and improves the codegen for inserts involving float.

Handling for `Create(Vector128<T> lower, Vector128<T> upper)` also needs to be done, but will be a separate PR.

### CODE REVIEW

**Comment by tannergooding** [comment]
CC. @dotnet/jit-contrib

File: src/coreclr/jit/gentree.cpp
Comment (TIHan) [inline src/coreclr/jit/gentree.cpp:17342]: Should we always zero out `simd32Val`, even if it's not in the path of `Vector_Create` ?

Comment (tannergooding) [reply src/coreclr/jit/gentree.cpp:17342]: We could, but the value isn't expected to be used if `false` and so its extra memory operations that are technically "unnecessary"

File: src/coreclr/jit/lowerxarch.cpp
Comment (TIHan) [inline src/coreclr/jit/lowerxarch.cpp:7159]: I've seen this pattern of using `SetRegOptional` if a node cannot be contained - what is the difference between using `SetRegOptional `and marking the node as contained?

Comment (SingleAccretion) [reply src/coreclr/jit/lowerxarch.cpp:7159]: "RegOptional" (aka "register optional") means the node _may_ end up effectively contained (used from memory: either a local, or a spill temp). Generally, all operations that support memory containment also support `RegOptional`.

That said, this particular pattern is not correct. `RegOptional` has the same use semantics as containment, i. e. it can only be set if we know the node is safe to contain (in the `IsSafeToContainMem` sense).

Comment (tannergooding) [reply src/coreclr/jit/lowerxarch.cpp:7159]: > That said, this particular pattern is not correct. RegOptional has the same use semantics as containment, i. e. it can only be set if we know the node is safe to contain (in the IsSafeToContainMem sense).

That should be getting appropriately checked as part of `TryGetContainableHWIntrinsicOp`. If it isn't, its a fix in that method and not a fix here.

Comment (SingleAccretion) [reply src/coreclr/jit/lowerxarch.cpp:7159]: > If it isn't, its a fix in that method and not a fix here.

Yes, and I do see it isn't checked appropriately in `TryGetContainableHWIntrinsicOp` (it is always set for `supportsGeneralLoads`).

(Was commenting on that mostly because there is a common alternation of this: `if (IsContainableMemOp() && IsSafeToContainMem(...) { MakeSrcContained(); } else { SetRegOptional() }`)

Comment (tannergooding) [reply src/coreclr/jit/lowerxarch.cpp:7159]: Since this is a more general pre-existing issue, I'll get a fix up in another PR. Should be straightforward since it just requires checking `IsSafeToContainMem` before setting `supportsRegOptional`

**Comment by tannergooding** [comment]
`outerloop` failures are:
* https://github.com/dotnet/runtime/issues/77477
* https://github.com/dotnet/runtime/issues/77791
* https://github.com/dotnet/runtime/issues/77790

File: src/coreclr/jit/lowerxarch.cpp
Comment (tannergooding) [reply src/coreclr/jit/lowerxarch.cpp:7159]: Fix up https://github.com/dotnet/runtime/pull/77895

File: src/coreclr/jit/lowerxarch.cpp
Comment (EgorBo) [inline src/coreclr/jit/lowerxarch.cpp:1752]: So is it a correct way to zero-extend a small type? Isn't this creating a node with TYP_UBYTE in lower? cc @SingleAccretion. I though it supposed to be TYP_INT, op1, true, TYP_UBYTE

File: src/coreclr/jit/gentree.cpp
Comment (EgorBo) [inline src/coreclr/jit/gentree.cpp:18949]: I assume it's also true for Vector256? (Avx Insert). just for my understanding

File: src/coreclr/jit/lowerxarch.cpp
Comment (SingleAccretion) [reply src/coreclr/jit/lowerxarch.cpp:1752]: True, that looks like a typo. Should be `gtNewCastNode(TYP_INT, op1, /* fromUnsigned */ false, TYP_UBYTE)` (`GTF_UNSIGNED` not set to not confuse people unnecessarily).

**Review by EgorBo — APPROVED** [review]
LGTM, assuming the test failures aren't related modulo the GT_CAST question.
Also, I assume it doesn't break mono since it converts a helper method into a "recursive" intrinsic

File: src/coreclr/jit/gentree.cpp
Comment (tannergooding) [reply src/coreclr/jit/gentree.cpp:18949]: There is no `Insert` for `Avx/Avx2`, only `InsertVector128`.

File: src/coreclr/jit/lowerxarch.cpp
Comment (tannergooding) [reply src/coreclr/jit/lowerxarch.cpp]: Fixed. I really don't like how we handle casts for small types. It is especially painful for HWIntrinsics since we need to actually deal with small types and consume them directly as small types.

**Comment by tannergooding** [comment]
CI failures are currently https://github.com/dotnet/arcade/issues/11554. However, there is no way to tell if its masking actual issues or not so I'll hold off on merging.

**Pull request merged**
